plugins/org.eclipse.n4js.xtext.ide/src/org/eclipse/n4js/xtext/ide/server/issues/DiagnosticComparator.java

```diff
@@ -37,8 +37,10 @@
 						d1.getSeverity() != null ? d1.getSeverity() : DiagnosticSeverity.Hint,
 						d2.getSeverity() != null ? d2.getSeverity() : DiagnosticSeverity.Hint)
 				.compare(
-						d1.getCode() != null ? d1.getCode() : "",
-						d2.getCode() != null ? d2.getCode() : "")
+						d1.getCode() != null ? d1.getCode().getLeft() != null ? d1.getCode().getLeft()
+								: d1.getCode().getRight().intValue() : "",
```

Comment:
The implementation of class `Either` seems to assume that `left` and `right` may both be `null` (for example see the implementation of `Either#hashCode()`) so we should probably guard against this here as well to avoid a null pointer exception at `...getRight().intValue()`.

Reply:
So you mean we create diagnostics without haven a value for `code`?

Reply:
What I meant was: `d1.getCode().getRight()` could return `null` here, leading to a NullPointerException.

Reply:
Ok, I see. 👍